src/GitVersion.Output.Tests/Output/FormatArgumentTests.cs

```diff
@@ -68,6 +68,126 @@ public void ShouldOutputFormatWithEnvironmentVariablesTests(string format, strin
+        var output = consoleBuilder.ToString();
+        output.ShouldContain("GitVersion_" + variableName + "=" + expectedValue + "\n");
+    }
+
+    [TestCase]
+    public void ShouldOutputAllCalculatedVariablesAsDotEnvEntries()
+    {
+        var fixture = CreateTestRepository();
+
+        var consoleBuilder = new StringBuilder();
+        IConsole consoleAdapter = new TestConsoleAdapter(consoleBuilder);
+
+        var sp = ConfigureServices(services =>
+        {
+            var options = Options.Create(new GitVersionOptions { WorkingDirectory = fixture.RepositoryPath, RepositoryInfo = { TargetBranch = fixture.Repository.Head.CanonicalName }, Output = { OutputType.DotEnv } });
+            var repository = fixture.Repository.ToGitRepository();
+
+            services.AddSingleton(options);
+            services.AddSingleton(repository);
+            services.AddSingleton(consoleAdapter);
+        });
+
+        var versionVariables = sp.GetRequiredService<IGitVersionCalculateTool>().CalculateVersionVariables();
+        var outputGenerator = sp.GetRequiredService<IOutputGenerator>();
+
+        outputGenerator.Execute(versionVariables, new());
+        var output = consoleBuilder.ToString();
+        var totalOutputLines = output.Split("\n").Length - 1; // ignore last item that also ends with \n
+        Assert.That(totalOutputLines, Is.EqualTo(versionVariables.Count()));
+    }
+
+    [TestCase("Major", "0")]
+    [TestCase("MajorMinorPatch", "0.0.1")]
+    [TestCase("SemVer", "0.0.1-1")]
+    [TestCase("BuildMetaData", "''")]
+    [TestCase("AssemblySemVer", "0.0.1.0")]
+    [TestCase("PreReleaseTagWithDash", "-1")]
+    [TestCase("BranchName", "main")]
+    [TestCase("PreReleaseLabel", "''")]
+    [TestCase("PreReleaseLabelWithDash", "''")]
+    public void ShouldOutputAllDotEnvEntriesEvenForMinimalRepositories(string variableName, string expectedValue)
+    {
+        var fixture = CreateMinimalTestRepository();
+
+        var consoleBuilder = new StringBuilder();
+        IConsole consoleAdapter = new TestConsoleAdapter(consoleBuilder);
+
+        var sp = ConfigureServices(services =>
+        {
+            var options = Options.Create(new GitVersionOptions { WorkingDirectory = fixture.RepositoryPath, RepositoryInfo = { TargetBranch = fixture.Repository.Head.CanonicalName }, Output = { OutputType.DotEnv } });
+            var repository = fixture.Repository.ToGitRepository();
+
+            services.AddSingleton(options);
+            services.AddSingleton(repository);
+            services.AddSingleton(consoleAdapter);
+        });
+
+        var versionVariables = sp.GetRequiredService<IGitVersionCalculateTool>().CalculateVersionVariables();
+        var outputGenerator = sp.GetRequiredService<IOutputGenerator>();
+
+        outputGenerator.Execute(versionVariables, new());
+        var output = consoleBuilder.ToString();
+        output.ShouldContain("GitVersion_" + variableName + "=" + expectedValue + "\n");
```

Comment:
Use `Environment.NewLine` instead of `\n` for increased cross-platform compatibility. String interpolation may increase readability here too.

```suggestion
        output.ShouldContain($"GitVersion_{variableName}={expectedValue}{Environment.NewLine}");
```